internal/impl/packages.go

```diff
@@ -214,34 +217,21 @@ func (d *Devbox) removePackagesFromProfile(ctx context.Context, pkgs []string) e
 		return err
 	}
 
-	items, err := nix.ProfileListItems(d.writer, profileDir)
-	if err != nil {
-		return err
-	}
-
-	nameToAttributePath := map[string]string{}
-	for _, item := range items {
-		attrPath, err := item.AttributePath()
-		if err != nil {
-			return err
-		}
-		name, err := item.PackageName()
-		if err != nil {
-			return err
-		}
-		nameToAttributePath[name] = attrPath
-	}
-
 	for _, pkg := range pkgs {
-		attrPath, ok := nameToAttributePath[pkg]
-		if !ok {
-			return errors.Errorf("Did not find AttributePath for package: %s", pkg)
+		index, err := nix.ProfileListIndex(&nix.ProfileListIndexArgs{
+			Writer:     d.writer,
+			Pkg:        pkg,
+			ProjectDir: d.projectDir,
+			ProfileDir: profileDir,
+		})
+		if err != nil {
+			ux.Ferror(d.writer, "Package %s not found in profile. Skipping.\n", pkg)
 		}
 
 		// TODO: unify this with nix.ProfileRemove
 		cmd := exec.Command("nix", "profile", "remove",
 			"--profile", profileDir,
-			attrPath,
+			fmt.Sprintf("%d", index),
```

Comment:
using index instead of (packageName, attributePath) sounds good

Could you double-check that the index is stable? Or at least stable enough for the purposes of these operations?

For example: what happens if I am installing 3 packages at the same time?

Reply:
It's stable through adding but not removing. Which is why I added the ability to pass a pre-computed list. When we are removing we recompute the list every time. When adding we compute it once and then add all packages. 

Computing the list every time is not very slow, but we don't want it to happen on `ensure` since that gets called on every shell, run etc. 

In a follow up, we could even cache the list in a file which would allow a 20-30ms speed-up of ensure.

Reply:
I don't know if this is the biggest perf bottleneck. 20-30 ms doesn't seem that much.